iib/web/errors.py

```diff
@@ -32,3 +35,34 @@ def json_error(error):
         response = jsonify({'error': msg})
         response.status_code = status_code
     return response
+
+
+def handle_broker_error(iib_request):
+    """
+    Catches RabbitMQ error and raises IIBError instead.
+
+    :param Request iib_request: type of an index image build request
+    """
+
+    def get_decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except kombu.exceptions.OperationalError:
+                payload = request.get_json()
+                req = iib_request.from_json(payload)
```

Comment:
This actually creates a new `Request` object, so you'll end up with duplicate requests. One will still be `in_progress` and the other will be `failed`. You need to filter for the request that was just created and failed in this request.